File: ruby_SDK/PayHubSDK/lib/PayHubSDK/com/payhub/ws/api/schedule_information.rb
class ScheduleInformation< AbstractInfo
  include JsonSerializer
  attr_accessor :schedule
  # @param [Object] transactionManager
  def initialize(transactionManager=nil)
    @transactionManager=transactionManager
    @transactionType=TransactionType::Schedule
  end

  def convertData(json)
    @schedule = Schedule.from_json(json)
  end

  def getUrlForTransactionType(type)
    url=nil
    if(TransactionType::Sale==(type))
      url =  "sale/"
    end
    if(TransactionType::AuthOnly==(type))
      url =  "authonly/"
    end
    if(TransactionType::Capture==(type))
      url =  "capture/"
    end
    if(TransactionType::Bill==(type))
      url =  "bill"
    end
    if(TransactionType::CardData==(type))
      url =  "carddata/"
    end
    if(TransactionType::Customer==(type))
      url =  "customer/"
    end
    if(TransactionType::Merchant==(type))
      url =  "merchant/"
    end
    if(TransactionType::RecurringBill==(type))
      url =  "recurring-bill/"
    end
    if(TransactionType::Schedule==(type))
      url =  "schedule/"
    end
    if(TransactionType::Refund==(type))
      url =  "refund/"
    end
    if(TransactionType::VoidTransaction==(type))
      url =  "void/"
    end
    if(TransactionType::Verify==(type))
      url =  "verify/"
    end
    return url
  end

end

File: Payhub - SDK/php_SDK/com/payhub/ws/api/ScheduleInformation.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 24/07/2015
 * Time: 11:05
 */
class ScheduleInformation extends AbstractInfo
{
    private $schedule;
    /**
     * RefundInformation constructor.
     * @param $transactionManager
     */
    public function __construct($transactionManager)
    {
        if(!is_null($transactionManager)) {
            $this->transactionManager = $transactionManager;
        }
        $this->transactionType=TransactionType::Schedule;
    }
    /**
     * @return mixed
     */
    public function getSchedule()
    {
        return $this->schedule;
    }

    /**
     * @param mixed $schedule
     */
    public function setSchedule($schedule)
    {
        $this->schedule = $schedule;
    }

    public function convertData($json)
    {
        $data = json_decode($json, true);
        $this->schedule = Schedule::fromArray($data);
    }

    public function getUrlForTransactionType($type)
    {
        $url=null;
        if(TransactionType::Sale==($type)){
            $url =  "sale/";
        }
        if(TransactionType::AuthOnly==($type)){
            $url =  "authonly/";
        }
        if(TransactionType::Capture==($type)){
            $url =  "capture/";
        }if(TransactionType::Bill==($type)){
        $url =  "bill";
        }if(TransactionType::CardData==($type)){
            $url =  "carddata/";
        }if(TransactionType::Customer==($type)){
            $url =  "customer/";
        }if(TransactionType::Merchant==($type)){
            $url =  "merchant/";
        }if(TransactionType::RecurringBill==($type)){
            $url =  "recurring-bill/";
        }if(TransactionType::Schedule==($type)){
            $url =  "schedule/";
        }if(TransactionType::Refund==($type)){
            $url =  "refund/";
        }if(TransactionType::VoidTransaction==($type)){
            $url =  "void/";
        }
        if(TransactionType::Verify==($type)){
            $url =  "verify/";
        }
        return $url;
    }
}